modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/resource/v1_0/ProductConfigurationListAccountResourceImpl.java

```diff
@@ -0,0 +1,266 @@
+import com.liferay.headless.commerce.admin.catalog.dto.v1_0.ProductConfigurationList;
+import com.liferay.headless.commerce.admin.catalog.dto.v1_0.ProductConfigurationListAccount;
+import com.liferay.headless.commerce.admin.catalog.resource.v1_0.ProductConfigurationListAccountResource;
+import com.liferay.portal.kernel.search.Sort;
+import com.liferay.portal.kernel.search.filter.Filter;
+import com.liferay.portal.kernel.security.permission.resource.ModelResourcePermission;
+import com.liferay.portal.kernel.util.HashMapBuilder;
+import com.liferay.portal.kernel.util.Validator;
+import com.liferay.portal.vulcan.dto.converter.DTOConverter;
+import com.liferay.portal.vulcan.dto.converter.DTOConverterRegistry;
+import com.liferay.portal.vulcan.dto.converter.DefaultDTOConverterContext;
+import com.liferay.portal.vulcan.fields.NestedField;
+import com.liferay.portal.vulcan.pagination.Page;
+import com.liferay.portal.vulcan.pagination.Pagination;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.component.annotations.ServiceScope;
+
+/**
+ * @author Danny Situ
+ */
+@Component(
+	properties = "OSGI-INF/liferay/rest/v1_0/product-configuration-list-account.properties",
+	property = "nested.field.support=true", scope = ServiceScope.PROTOTYPE,
+	service = ProductConfigurationListAccountResource.class
+)
+public class ProductConfigurationListAccountResourceImpl
+	extends BaseProductConfigurationListAccountResourceImpl {
+
+	@Override
+	public void deleteProductConfigurationListAccount(Long id)
+		throws Exception {
+
+		_cpConfigurationListRelService.deleteCPConfigurationListRel(id);
+	}
+
+	@Override
+	public Page<ProductConfigurationListAccount>
+			getProductConfigurationListByExternalReferenceCodeProductConfigurationListAccountsPage(
+				String externalReferenceCode, Pagination pagination)
+		throws Exception {
+
+		CPConfigurationList cpConfigurationList =
+			_cpConfigurationListService.
+				fetchCPConfigurationListByExternalReferenceCode(
+					externalReferenceCode, contextCompany.getCompanyId());
+
+		if (cpConfigurationList == null) {
+			throw new NoSuchCPConfigurationListException(
+				"Unable to find product configuration list with external " +
+					"reference code " + externalReferenceCode);
+		}
+
+		List<CPConfigurationListRel> cpConfigurationListRels =
+			_cpConfigurationListRelService.getCPConfigurationListRels(
+				AccountEntry.class.getName(),
+				cpConfigurationList.getCPConfigurationListId(),
+				pagination.getStartPosition(), pagination.getEndPosition(),
+				null);
+
+		int totalItems =
+			_cpConfigurationListRelService.getCPConfigurationListRelsCount(
+				AccountEntry.class.getName(),
+				cpConfigurationList.getCPConfigurationListId());
+
+		return Page.of(
+			_toProductConfigurationListAccounts(cpConfigurationListRels),
+			pagination, totalItems);
```

Comment:
please call the other method so that you don't repeat the logic + you can inline directly the service call in the Pagination constructor method call